code/modules/antagonists/wizard/equipment/artefact.dm

```diff
@@ -299,78 +299,80 @@
-		end_effect(user)
-		return
-	var/breakout = 0
-	while(breakout < 50)
-		var/turf/potential_T = find_safe_turf()
-		if(T.z != potential_T.z || abs(get_dist_euclidian(potential_T,T)) > 50 - breakout)
-			do_teleport(user, potential_T, channel = TELEPORT_CHANNEL_MAGIC)
-			T = potential_T
-			break
-		breakout += 1
-	new /obj/effect/temp_visual/tornado(T)
-	sleep(20)
-	end_effect(user)
-	if(interrupted(user))
+	if(whistler)
+		to_chat(user, span_warning("[src] is already warping."))
 		return
-	on_cooldown = 2
-	addtimer(VARSET_CALLBACK(src, on_cooldown, 0), 4 SECONDS)
 
-/obj/item/warpwhistle/Destroy()
-	if(on_cooldown == 1 && last_user) //Flute got dunked somewhere in the teleport
-		end_effect(last_user)
-	return ..()
+	whistler = user
+	var/turf/current_turf = get_turf(user)
+	var/turf/spawn_location = locate(user.x + pick(-7, 7), user.y, user.z)
+	playsound(current_turf,'sound/magic/warpwhistle.ogg', 200, TRUE)
+	new /obj/effect/temp_visual/teleporting_tornado(spawn_location, src)
+	COOLDOWN_START(src, whistle_cooldown, 4 SECONDS)
 
-/obj/effect/temp_visual/tornado
-	icon = 'icons/obj/wizard.dmi'
-	icon_state = "tornado"
+///Teleporting tornado, spawned by warp whistle, teleports the user if they manage to pick them up.
+/obj/effect/temp_visual/teleporting_tornado
 	name = "tornado"
 	desc = "This thing sucks!"
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "tornado"
 	layer = FLY_LAYER
 	plane = ABOVE_GAME_PLANE
-	randomdir = 0
-	duration = 40
-	pixel_x = 500
+	randomdir = FALSE
+	duration = 3 SECONDS
+	movement_type = PHASING
+
+	/// Reference to the whistle
+	var/obj/item/warp_whistle/whistle
+	/// List of all mobs currently held by the tornado.
+	var/list/pickedup_mobs = list()
 
-/obj/effect/temp_visual/tornado/Initialize(mapload)
+/obj/effect/temp_visual/teleporting_tornado/Initialize(mapload, obj/item/warp_whistle/whistle)
 	. = ..()
-	animate(src, pixel_x = -500, time = 40)
+	src.whistle = whistle
+	if(!whistle)
+		qdel(src)
+		return
+	RegisterSignal(src, COMSIG_MOVABLE_CROSS_OVER, .proc/check_teleport)
+	SSmove_manager.move_towards(src, get_turf(whistle.whistler))
+
+/// Check if anything the tornado crosses is the creator.
+/obj/effect/temp_visual/teleporting_tornado/proc/check_teleport(datum/source, atom/movable/crossed)
+	SIGNAL_HANDLER
+	if(crossed != whistle.whistler)
+		return
+
+	pickedup_mobs += crossed
+	buckle_mob(crossed, TRUE, FALSE)
+	animate(src, alpha = 20, pixel_y = 400, time = 3 SECONDS)
+	animate(crossed, pixel_y = 400, time = 3 SECONDS)
+
+/// Destroy the tornado and teleport everyone on it away.
+/obj/effect/temp_visual/teleporting_tornado/Destroy()
+	if(!isnull(pickedup_mobs))
+		var/turf/ending_turfs = find_safe_turf()
+		for(var/mob/stored_mobs as anything in pickedup_mobs)
+			do_teleport(stored_mobs, ending_turfs, channel = TELEPORT_CHANNEL_MAGIC)
+			animate(stored_mobs, pixel_y = null, time = 1 SECONDS)
+			stored_mobs.log_message("warped with [whistle].", LOG_ATTACK, color = "red")
```

Comment:
Do we rely on this? or is it just a side effect

Reply:
The logging? I added it in because I thought it would be useful for admins to know what happened to someone mid-arrest to get away

Reply:
I believe they're referring to the `do_teleport()` in `Destroy()`

Reply:
I want the players to be there while the tornado is flying away with their body while still respecting only being around for 3 seconds, this was the easiest way I thought to make that possible.

Reply:
rather then a qdel_in(3 seconds), why not queue a proc that moves the player, and then deletes the tornado
I'd like to keep Destroy as focused on resultless cleanup as possible, rather then having it do behavior like this